sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client.c

```diff
@@ -75,60 +91,162 @@ static void test_az_iot_provisioning_client_custom_options_get_username_succeed(
 
-static void test_az_iot_provisioning_client_get_subscribe_topic_filter_succeed()
+static void test_az_iot_provisioning_client_get_connect_info_insufficient_space_fails()
 {
   az_iot_provisioning_client client;
-  char topic[128];
-  memset(topic, 0xCC, sizeof(topic));
+  az_iot_provisioning_client_options options = az_iot_provisioning_client_options_default();
+  options.user_agent = AZ_SPAN_FROM_STR(TEST_USER_AGENT);
+
+  az_result ret = az_iot_provisioning_client_init(
+      &client,
+      test_global_device_endpoint,
+      AZ_SPAN_FROM_STR(TEST_ID_SCOPE),
+      AZ_SPAN_FROM_STR(TEST_REGISTRATION_ID),
+      &options);
+  assert_int_equal(AZ_OK, ret);
 
-  char* expected_topic = "$dps/registrations/res/#";
+  char client_id[sizeof(TEST_REGISTRATION_ID) - 1];
+  memset(client_id, 0xCC, sizeof(client_id));
+  size_t client_id_len = 0xBAADC0DE;
+  ret = az_iot_provisioning_client_get_client_id(
+      &client, client_id, sizeof(client_id), &client_id_len);
+  assert_int_equal(AZ_ERROR_INSUFFICIENT_SPAN_SIZE, ret);
+  for (size_t i = 0; i < sizeof(client_id); i++)
+  {
+    assert_int_equal((uint8_t)0xCC, (uint8_t)client_id[i]);
+  }
+
+  assert_int_equal(0xBAADC0DE, client_id_len);
+
+  char expected_username[] = TEST_ID_SCOPE
+      "/registrations/" TEST_REGISTRATION_ID
+      "/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION USER_AGENT_PREFIX TEST_USER_AGENT;
 
+  char user_name[sizeof(expected_username) - 1];
+  memset(user_name, 0xCC, sizeof(user_name));
+  size_t user_name_len = 0xBAADC0DE;
+  ret = az_iot_provisioning_client_get_user_name(
+      &client, user_name, sizeof(user_name), &user_name_len);
+  assert_int_equal(AZ_ERROR_INSUFFICIENT_SPAN_SIZE, ret);
+  for (size_t i = 0; i < sizeof(user_name); i++)
+  {
+    assert_int_equal((uint8_t)0xCC, (uint8_t)user_name[i]);
+  }
+
+  assert_int_equal(0xBAADC0DE, user_name_len);
+}
+
+static void test_az_iot_provisioning_client_get_subscribe_topic_filter_succeed()
+{
+  az_iot_provisioning_client client;
+  char expected_topic[] = "$dps/registrations/res/#";
+
+  char topic[sizeof(expected_topic) + 1];
+  memset(topic, 0xCC, sizeof(topic));
   az_result ret = az_iot_provisioning_client_register_get_subscribe_topic_filter(
       &client, topic, sizeof(topic), NULL);
 
   assert_int_equal(AZ_OK, ret);
   assert_memory_equal(expected_topic, topic, strlen(topic));
   assert_int_equal((uint8_t)0xCC, (uint8_t)topic[strlen(topic) + 1]);
+
+  size_t topic_len;
+  ret = az_iot_provisioning_client_register_get_subscribe_topic_filter(
+      &client, topic, sizeof(topic), &topic_len);
+  assert_int_equal(AZ_OK, ret);
+  assert_memory_equal(expected_topic, topic, strlen(topic));
+  assert_int_equal((uint8_t)0xCC, (uint8_t)topic[strlen(topic) + 1]);
+  assert_int_equal(strlen(expected_topic), topic_len);
 }
 
-static void test_az_iot_provisioning_client_get_register_publish_topic_filter_succeed()
+static void test_az_iot_provisioning_client_get_subscribe_topic_filter_insufficient_space_fails()
```

Comment:
Lol, such long test names really makes you miss objects/instances and nesting from other languages :)

Reply:
Yeah and I can't easily read the unit, action and expected results either. Maybe we change to some other naming convention?